pkg/build/generator/generator.go

```diff
@@ -513,6 +513,11 @@ func (g *BuildGenerator) generateBuildFromConfig(ctx kapi.Context, bc *buildapi.
 				Kind: "DockerImage",
 				Name: runtimeImageName,
 			}
+			// We're using single secret for pulling builder and runtime images.
+			// If builder image doesn't need a secret, let's check whether runtime image needs it or not.
+			if build.Spec.Strategy.SourceStrategy.PullSecret == nil {
+				build.Spec.Strategy.SourceStrategy.PullSecret = g.resolveImageSecret(ctx, builderSecrets, build.Spec.Strategy.SourceStrategy.RuntimeImage, bc.Namespace)
+			}
```

Comment:
I thought when @PI-Victor and I discussed this, the plan was that resolveImageSecret was going to look for a secret that included credentials for both the builder image and runtime image, and return that.

In the case that one of the two images doesn't need a secret (which will basically mean you can't find a secret that matches it), then we should fall back to using a secret that at least matches one of the images (prefer builder).

In otherwords, if I have the following secrets available:
1) creds[builderPull,runtimePull]
2) creds[builderPull]
3) creds[runtimePull]

We should prefer them in that order.

What you're doing here is similar, but not identical, since it's possible that when we looked for the builder secret, we found a secret that has the builder pull auth but does not have the runtime pull auth, and there might be a second secret that has auths for both.